Server/ReasnAPI/ReasnAPI/Services/UserService.cs

```diff
@@ -1,16 +1,292 @@
-﻿using ReasnAPI.Models.Database;
-
-namespace ReasnAPI.Services {
-    public class UserService (ReasnContext context) {
-        private readonly ReasnContext _context = context;
-
-        /* TODO: Create following functions for this class
-         * create
-         * update
-         * delete
-         * get by ID
-         * get list by filter
-         * get all
-         */
-    }
-}
+using ReasnAPI.Models.Database;
+using ReasnAPI.Models.DTOs;
+using System.Linq.Expressions;
+using System.Transactions;
+
+namespace ReasnAPI.Services
+{
+    public class UserService(ReasnContext context)
+    {
+        public UserDto? UpdateUser(int userId, UserDto? userDto)
+        {
+            using (var scope = new TransactionScope())
+            {
+                if (userDto is null)
+                {
+                    return null;
+                }
+
+                var user = context.Users.FirstOrDefault(r => r.Id == userId);
+
+                if (user is null)
+                {
+                    return null;
+                }
+
+                var usernameExists = context.Users.Any(r => r.Username == userDto.Username && r.Id != userId);
+
+                if (usernameExists)
+                {
+                    return null;
+                }
+
+                user.Username = userDto.Username;
+                user.Name = userDto.Name;
+                user.Surname = userDto.Surname;
+                user.Username = userDto.Username;
+                user.Email = userDto.Email;
+                user.Phone = userDto.Phone;
+                user.Role = userDto.Role;
+                user.AddressId = userDto.AddressId;
+                user.UpdatedAt = DateTime.UtcNow;
+
+                context.Users.Update(user);
+
+                if (userDto.Interests is null || userDto.Interests.Count == 0)

```

Comment:
```suggestion
                if (userDto.Interests is null || !userDto.Interests.Any())

```

Reply:
Visual Studio suggest to leave it as it is for clarity. I also think `.Count == 0` is easier to read.